packages/x-components/src/components/preselected-filters.vue

```diff
@@ -0,0 +1,106 @@
+<script lang="ts">
+  import { Component, Inject, Prop } from 'vue-property-decorator';
+  import Vue from 'vue';
+  import { createRawFilters } from '../utils/filters';
+  import { isArrayEmpty } from '../utils/array';
+  import { SnippetConfig } from '../x-installer/api/api.types';
+
+  /**
+   * This component emits {@link XEventsTypes.PreselectedFiltersProvided} when a preselected filter
+   * is set in the snippet config or by using the prop of the component.
+   *
+   * @public
+   */
+  @Component
+  export default class PreselectedFilters extends Vue {
+    /**
+     * Injects {@link SnippetConfig} provided by an ancestor as snippetConfig.
+     *
+     * @internal
+     */
+    @Inject()
+    public snippetConfig?: SnippetConfig;
+
+    /**
+     * A list of preselected filters.
+     *
+     * @remarks Emits the {@link XEventsTypes.PreselectedFiltersProvided} when the
+     * component is rendered or the values changed.
```

Comment:
This says that it emits the event whenever the value changes, but it is not true. This is done only once in the `mounted` event, without any watcher or XEmit decorator. 

This is a use case for dutti which wants to change this filters dynamically. Maybe it can be postponed until someone requests it. I don't know if Vodafone needs this.